framework/src/Volo.Abp.Ddd.Application/Volo/Abp/Application/Services/AbstractKeyCrudAppService.cs

```diff
@@ -62,6 +62,8 @@ public abstract class AbstractKeyCrudAppService<TEntity, TGetOutputDto, TGetList
             ICrudAppService<TGetOutputDto, TGetListOutputDto, TKey, TGetListInput, TCreateInput, TUpdateInput>
         where TEntity : class, IEntity
     {
+        protected new IRepository<TEntity> Repository { get; }
```

Comment:
How about this?
This saves us double property assignments.

```csharp
protected new IRepository<TEntity> Repository => (IRepository<TEntity>)base.Repository;
```

Reply:
`protected new` doesn't seem to be a good way, I think we can use different names.

```cs
 protected IReadOnlyRepository<TEntity> ReadOnlyRepository { get; }

 protected IRepository<TEntity> Repository { get; }

```

Reply:
Okay, I'll update the property assignments.